lib/commands/performance.js

```diff
@@ -113,73 +115,81 @@ helpers.getMemoryInfo = async function getMemoryInfo (packageName, dataReadTimeo
     let data = await this.adb.shell(cmd);
     if (!data) throw new Error('No data from dumpsys'); //eslint-disable-line curly
 
-    let totalPrivateDirty, totalPss,
-        nativePrivateDirty, nativePss, nativeHeapSize, nativeHeapAllocatedSize,
-        dalvikPrivateDirty, dalvikPss,
+    let totalPrivateDirty, totalPss, totalRss,
+        nativePrivateDirty, nativePss, nativeHeapSize, nativeHeapAllocatedSize, nativeRss,
+        dalvikPrivateDirty, dalvikPss, dalvikRss,
         eglPrivateDirty, eglPss,
         glPrivateDirty, glPss;
     let apilevel = await this.adb.getApiLevel();
     for (let line of data.split('\n')) {
       let entries = line.trim().split(' ').filter(Boolean);
-      // entries will have the values
+      // API level 30 and above
+      //   ['<System Type>', '<Memory Type>', <pss total>, <private dirty>, <private clean>, <swapPss dirty>, <rss total>, <heap size>, <heap alloc>, <heap free>]
+      //
+      // API level between 30 and 18
       //   ['<System Type>', '<Memory Type>', <pss total>, <private dirty>, <private clean>, <swapPss dirty>, <heap size>, <heap alloc>, <heap free>]
       // except 'TOTAL', which skips the second type name
       //
       // and on API level 18 and below
       //   ['<System Type', '<pps>', '<shared dirty>', '<private dirty>', '<heap size>', '<heap alloc>', '<heap free>']
 
-      if (apilevel > 18) {
-        let type = entries[0];
-        let subType = entries[1];
+      if (apilevel > 18 && apilevel < 30) {
+        const [type, subType] = entries;
         if (type === 'Native' && subType === 'Heap') {
           // native heap
-          nativePss = entries[2];
-          nativePrivateDirty = entries[3];
-          nativeHeapSize = entries[6];
-          nativeHeapAllocatedSize = entries[7];
+          [,, nativePss, nativePrivateDirty,,, nativeHeapSize, nativeHeapAllocatedSize] = entries;
         } else if (type === 'Dalvik' && subType === 'Heap') {
           // dalvik heap
-          dalvikPss = entries[2];
-          dalvikPrivateDirty = entries[3];
+          [,, dalvikPss, dalvikPrivateDirty] = entries;
         } else if (type === 'EGL' && subType === 'mtrack') {
           // egl
-          eglPss = entries[2];
-          eglPrivateDirty = entries[3];
+          [,, eglPss, eglPrivateDirty] = entries;
         } else if (type === 'GL' && subType === 'mtrack') {
           // gl
-          glPss = entries[2];
-          glPrivateDirty = entries[3];
+          [,, glPss, glPrivateDirty] = entries;
         } else if (type === 'TOTAL' && entries.length === 8) {
           // there are two totals, and we only want the full listing, which has 8 entries
-          totalPss = entries[1];
-          totalPrivateDirty = entries[2];
+          [, totalPss, totalPrivateDirty] = entries;
+        }
+      } else if (apilevel >= 30) {
+        const [type, subType] = entries;
```

Comment:
it might be better readable if we move data parsers for different API versions to helper functions. Also, I would prefer to have magic strings like 'GL' or 'TOTAL' kept in constants.